### ISSUE TITLE
Make the code follow PEP 8 standard

### ISSUE THREAD

**Issue by dinkopehar** [eagleEggs/EHX#1]
I'm not commenting on reddit, but here instead. You should read more about PEP 8 and make your code follow it. Although you created this, you code has a lot of errors that don't follow it and it's not readable.

Here is the link to [PEP 8](https://pep8.org). And here is [cheetsheat ](https://gist.github.com/RichardBronosky/454964087739a449da04)made in code if you don't want to read all that.

**Comment by eagleEggs** [comment]
Good point. I will get this completed on the next version and make this a part of the process before releases going forward.

**Comment by eagleEggs** [comment]
I made PEP 8 changes for everything except the broad exceptions, which I'm putting something together for. Some of these longer strings also look a bit wonky when following the rules but I'm looking into that as well. It's in the PEP 8 branch if you have any other comments on it: https://github.com/eagleEggs/EHX/blob/PEP-8/EHX.py

Thanks for your feedback. It was interesting to look into PEP 8 and actually easier to read afterword,  I agree. Feels cleaner... it still has work to be done though.

**Comment by dinkopehar** [comment]
I made last edits. Added blank line at the and removed too much whitespace around ##GUI##. One more thing. Methods `stylize` and `stylize_remove` have parameter _parent_ that is not used ? If you are not using _parent_, you could remove it. That is first complaint I get when I clone your EHX.py .

EDIT:

After more examination and running, I get some wierd errors when trying to select some elements by class. For example, I open this web site https://fpmoz.sum.ba/index.php?option=com_content&view=category&layout=blog&id=14&Itemid=122&lang=hr . And it has a lot of class `row` . But it highlights some of them, and stops running and gives me this error:

```
C:\Users\Dine\envs\exercise_env\Scripts\python.exe C:/Users/Dine/Documents/Programming/WebScraping/bzv.py
<selenium.webdriver.remote.webelement.WebElement (session="686787b45ad33da24a43cb9bfd30f330", element="0.5057475985579427-1")>.setAttribute('style',background: Red; border: 3px solid Red;);
Traceback (most recent call last):
  File "C:/Users/Dine/Documents/Programming/WebScraping/bzv.py", line 163, in <module>
    "APP.highlight(LIGHT)".format(SCRIPTVAR))
  File "<string>", line 2, in <module>
  File "C:/Users/Dine/Documents/Programming/WebScraping/bzv.py", line 72, in highlight
    values["COLOR_TYPE"], values["COLOR_TYPE"]))
  File "C:/Users/Dine/Documents/Programming/WebScraping/bzv.py", line 82, in stylize
    " arguments[1]);", element, style)
  File "C:\Users\Dine\envs\exercise_env\lib\site-packages\selenium\webdriver\remote\webdriver.py", line 636, in execute_script
    'args': converted_args})['value']
  File "C:\Users\Dine\envs\exercise_env\lib\site-packages\selenium\webdriver\remote\webdriver.py", line 321, in execute
    self.error_handler.check_response(response)
  File "C:\Users\Dine\envs\exercise_env\lib\site-packages\selenium\webdriver\remote\errorhandler.py", line 242, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.WebDriverException: Message: unknown error: call function result missing 'value'
  (Session info: chrome=70.0.3538.102)
  (Driver info: chromedriver=2.33.506120 (e3e53437346286c0bc2d2dc9aa4915ba81d9023f),platform=Windows NT 10.0.17134 x86_64)

```

Am I missing something here ?

**Comment by eagleEggs** [comment]
Can you give me what you have in your element configuration? Also use the master branch as the PEP 8 branch needs more testing with the specific exceptions.

Also highlighting multiple elements at once (that all have the same tag) is something that is on the roadmap - As well as including more types other than CSS Selector, XPATH, and ID. These have not been tested currently.

Update: An update soon will support multiple elements (as well as classes specifically and more of the Selenium API for finding different element types). It's only coded to handle one now, but this is a good use case to build in.

**Comment by dinkopehar** [comment]
Nice. Sorry I was unresponsive. I think I cloned your pep 8 branch and runned a few times.  I opened a site in google chrome via 'rocket' icon, and than entered `.row` as CSS selector and clicked 'pen' and it marked a few of them and stopped. I tried to debug it, but I got lost in stacktrace.

Anyway, beside from that, you are on the right track 🥇

**Comment by eagleEggs** [comment]
Yep that's good. Easy enough. Will be in the next commit - Along with some much more granular exceptions which clean this up pretty well.

**Comment by eagleEggs** [comment]
The logic of this is almost mind boggling... But almost there :)

![image](https://user-images.githubusercontent.com/29800532/49058332-8b295200-f1d1-11e8-8bb5-2cbaa7f027b4.png)

It was never actually coded to do multiple elements of the same name, but it makes sense as that is pretty common and helps to visualize the site better when element hunting. Is definitely useful just from this test I did.

It would also not be entirely reliable if it only highlighted a single element, when there could be many more on the page with the same name without it showing you that in some way.

Good catch on this functionality.

I'm going to spin this off as a new issue to track it appropriately.

**Comment by dinkopehar** [comment]
Great 😸 . You are really on something great here xd. Maybe after some time, it will be amazing Desktop app for web scraping and "hunting" so to speak.

**Comment by eagleEggs** [comment]
Ok. Multiple Elements are now supported :) PEP 8 is about as good as it may get, and simplified the main loop into a single statement generator which should be slightly more performant as i got rid of the IF statement logic.

I will look into threading the call as you will notice there is a bit of lag when there are a lot of elements. It seems to hang but it will recover once it's finished processing.

... Ok I just finished implementing threading which speeds it up nicely :https://github.com/eagleEggs/EHX/tree/Threading

It's faster when threading the highlight over many elements but slower when removing them, which I'll get worked out. Also since highlighting many elements takes a bit longer I've added a pop up that goes away once the processing is complete.

I forgot to address the ‘parent’ issue. When I built these functions originally I didn’t include it because it seems to not require it. But it doesn’t work without it. I will look more into the root cause but it breaks functionality without it.

**Comment by eagleEggs** [comment]
Let me know how it goes! I'll close this issue Monday if there is nothing else you'd like to add and if it's all working for you :)

**Comment by dinkopehar** [comment]
Sorry, I wasn't able to check your code yet. I had job interview and project on my own. If I find some error or something to fix in your new code, I'll open the new issue. You can close this one.

Cheers 😄

**Comment by eagleEggs** [comment]
Great, hope you get the job and thanks much for your contribution and feedback on this project!!!
